purchases/src/test/java/com/revenuecat/purchases/DeviceCacheTest.kt

```diff
@@ -275,6 +277,89 @@ class DeviceCacheTest {
         verify { mockEditor.remove(cache.purchaserInfoCacheKey("legacyAppUserID")) }
     }
 
+    @Test
+    fun `clearing caches clears timestamps`() {
+        cache.setPurchaserInfoCacheTimestampToNow()
+        cache.setOfferingsCacheTimestampToNow()
+        mockString(cache.appUserIDCacheKey, "appUserID")
+        mockString(cache.legacyAppUserIDCacheKey, "legacyAppUserID")
+        mockString(cache.purchaserInfoCacheKey(appUserID), null)
+        cache.clearCachesForAppUserID()
+        assertThat(cache.isPurchaserInfoCacheStale()).isTrue()
+        assertThat(cache.isOfferingsCacheStale()).isTrue()
+    }
+
+    @Test
+    fun `invalidating offerings caches`() {
+        assertThat(cache.isOfferingsCacheStale()).isTrue()
+        cache.setOfferingsCacheTimestampToNow()
+        assertThat(cache.isOfferingsCacheStale()).isFalse()
+        cache.clearOfferingsCacheTimestamp()
+        assertThat(cache.isOfferingsCacheStale()).isTrue()
+    }
+
+    @Test
+    fun `stale if no caches`() {
+        assertThat(cache.isOfferingsCacheStale()).isTrue()
+        assertThat(cache.isPurchaserInfoCacheStale()).isTrue()
+    }
+
+    @Test
+    fun `purchaser info stale if longer than 5 minutes`() {
+        cache.cachePurchaserInfo("waldo", mockk(relaxed = true))
+        cache.purchaserInfoCachesLastUpdated = Date(0)
+        assertThat(cache.isPurchaserInfoCacheStale()).isTrue()
+        cache.purchaserInfoCachesLastUpdated = Date()
+        assertThat(cache.isPurchaserInfoCacheStale()).isFalse()
+    }
+
+    @Test
+    fun `offerings stale if longer than 5 minutes`() {
```

Comment:
this doesn't really test against 5 minutes, right? really, it'll succeed if and only if it passes in the value from the underlying object. 
so I'd call it `isCacheStale returns true if the cached object is stale` or something to that effect